TritonSE/ALUM-Mobile-Application
# Implements GET /mentor & /mentee + Auth Layer

### Tracking Info

Resolves #64 , #37 

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

- Added the get mentor/mentee route
- Added a get image route
- Added an auth layer to verify user credentials 

Note that the auth layer is something that will be used in all routes going forward. The create user route was updated to add custom roles to each user. This is our way of checking whether the user making a request is a mentor or mentee. 

Here is how the get route works on a high level:

1. Verify that the token is valid via Firebase API
2. Update the req body to include user type (Mentee or Mentor
3. For get Mentee...
4. If it is a mentee calling the route give all information, if its a mentor give only limited necessary information (see figma)
5. For get Mentor
6. If it is a mentor calling the route give all information, if its a mentee calling the route give limited information (see figma)
7. To get the image use the get image route

Something to also note is that the default ALUM image is stored once on the database. This is to optimize storage space. When creating a new user the image id will be set to the id of the default image stored in the database. This id will also be stored as an env variable. 

### Testing

How did you confirm your changes worked? 

I ran the commands on postman. I got the auth token using the post command that was sent on slack. I also tested the various edge cases which included auth errors and service errors.

### Confirmation of Change 

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

Get Image Route (for both mentor and mentee)

![Get image](https://user-images.githubusercontent.com/85322002/224915832-933b7c3c-4d08-4029-b3ba-3eb446402979.png)

Mentee get Mentee

![Mentor get Mentee](https://user-images.githubusercontent.com/85322002/224915905-eb4ad79d-bb86-4637-a536-d98f3d9de0db.png)

Mentee get Mentor

![Mentee get mentor](https://user-images.githubusercontent.com/85322002/224915965-b2fc28e6-4b30-4cef-8359-ae1805a2d9ac.png)

Mentor get Mentor

![Mentor get mentor](https://user-images.githubusercontent.com/85322002/224916000-ade465cf-adb1-4450-b54b-888bd141b11e.png)

Mentor get Mentee
![Mentor get Mentee](https://user-images.githubusercontent.com/85322002/224916031-0760c73a-fb75-4043-8a8f-486c9c4e5a8f.png)
